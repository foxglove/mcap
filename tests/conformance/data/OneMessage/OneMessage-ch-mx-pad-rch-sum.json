{
  "records": [
    {"type": "Header", "fields": [["library", ""], ["profile", ""]]},
    {
      "type": "Schema",
      "fields": [
        ["data", ["4", "5", "6"]],
        ["encoding", "c"],
        ["id", "1"],
        ["name", "Example"]
      ]
    },
    {
      "type": "Channel",
      "fields": [
        ["id", "1"],
        ["message_encoding", "a"],
        ["metadata", {"foo": "bar"}],
        ["schema_id", "1"],
        ["topic", "example"]
      ]
    },
    {
      "type": "Message",
      "fields": [
        ["channel_id", "1"],
        ["data", ["1", "2", "3"]],
        ["log_time", "2"],
        ["publish_time", "1"],
        ["sequence", "10"]
      ]
    },
    {"type": "DataEnd", "fields": [["data_section_crc", "670025892"]]},
    {
      "type": "Channel",
      "fields": [
        ["id", "1"],
        ["message_encoding", "a"],
        ["metadata", {"foo": "bar"}],
        ["schema_id", "1"],
        ["topic", "example"]
      ]
    },
    {
      "type": "SummaryOffset",
      "fields": [
        ["group_length", "50"],
        ["group_opcode", "4"],
        ["group_start", "242"]
      ]
    },
    {
      "type": "Footer",
      "fields": [
        ["summary_crc", "4117853642"],
        ["summary_offset_start", "292"],
        ["summary_start", "242"]
      ]
    }
  ],
  "meta": {"variant": {"features": ["ch", "mx", "rch", "sum", "pad"]}}
}
