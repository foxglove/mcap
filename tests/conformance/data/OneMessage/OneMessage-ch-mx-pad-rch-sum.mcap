version https://git-lfs.github.com/spec/v1
oid sha256:aab1980c012b7c2d14768ec6b8c643fc5b4d542da2daa5f0c45f69a54e90db32
size 358
