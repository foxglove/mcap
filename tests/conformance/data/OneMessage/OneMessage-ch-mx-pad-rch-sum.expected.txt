Header library=s:"" profile=s:""
Schema data=b:<040506> encoding=s:c id=i:1 name=s:Example
ChannelInfo id=i:1 message_encoding=s:a metadata=m:{s:foo=s:bar} schema_id=i:1 topic=s:example
Message channel_id=i:1 data=b:<010203> log_time=n:2 publish_time=n:1 sequence=i:10
MessageIndex channel_id=i:1 records=a:{n:2=n:81}
ChannelInfo id=i:1 message_encoding=s:a metadata=m:{s:foo=s:bar} schema_id=i:1 topic=s:example
ChunkIndex chunk_length=n:162 chunk_start_offset=n:28 compressed_size=n:115 compression=s:"" end_time=n:2 message_index_length=n:34 message_index_offsets=m:{i:1=n:192} start_time=n:2 uncompressed_size=n:115
SummaryOffset group_length=n:50 group_opcode=i:4 group_start=n:242
SummaryOffset group_length=n:0 group_opcode=i:13 group_start=n:292
SummaryOffset group_length=n:0 group_opcode=i:10 group_start=n:292
SummaryOffset group_length=n:90 group_opcode=i:8 group_start=n:292
Footer summary_crc=i:0 summary_offset_start=n:382 summary_start=n:242
